common/shared.c

```diff
@@ -836,18 +850,20 @@ ssize_t ft_post_inject(size_t size)
 {
 	ssize_t ret;
 
-	if (hints->caps & FI_TAGGED) {
-		ret = fi_tinject(ep, tx_buf, size + ft_tx_prefix_size(),
-				remote_fi_addr, tx_seq);
-	} else {
-		ret = fi_inject(ep, tx_buf, size + ft_tx_prefix_size(),
-				remote_fi_addr);
-	}
+	do {
+		if (hints->caps & FI_TAGGED) {
+			ret = fi_tinject(ep, tx_buf, size + ft_tx_prefix_size(),
+					remote_fi_addr, tx_seq);
+		} else {
+			ret = fi_inject(ep, tx_buf, size + ft_tx_prefix_size(),
+					remote_fi_addr);
+		}
+	} while (ret == -FI_EAGAIN);
+
```

Comment:
OK. Should the common function ft_inject be also updated to include a CQ/cntr read to guarantee progress?